File: complexcalcjava/MainActivity.java
package com.googleworkshop.jakub.complexcalc;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void add(View view){
        EditText et1 = (EditText)findViewById(R.id.editText);
        EditText et2 = (EditText)findViewById(R.id.editText2);

        ComplexNum c1 = new ComplexNum(et1.getText().toString());
        ComplexNum c2 = new ComplexNum(et2.getText().toString());

        TextView tv = (TextView)findViewById(R.id.textView5);
        tv.setText(c1.add(c2).toString());
    }

    public void subtract(View view){
        EditText et1 = (EditText)findViewById(R.id.editText);
        EditText et2 = (EditText)findViewById(R.id.editText2);

        ComplexNum c1 = new ComplexNum(et1.getText().toString());
        ComplexNum c2 = new ComplexNum(et2.getText().toString());

        TextView tv = (TextView)findViewById(R.id.textView5);
        tv.setText(c1.subtract(c2).toString());
    }

    public void multiply(View view){
        EditText et1 = (EditText)findViewById(R.id.editText);
        EditText et2 = (EditText)findViewById(R.id.editText2);

        ComplexNum c1 = new ComplexNum(et1.getText().toString());
        ComplexNum c2 = new ComplexNum(et2.getText().toString());

        TextView tv = (TextView)findViewById(R.id.textView5);
        tv.setText(c1.multiply(c2).toString());
    }

    public void divide(View view){
        EditText et1 = (EditText)findViewById(R.id.editText);
        EditText et2 = (EditText)findViewById(R.id.editText2);

        ComplexNum c1 = new ComplexNum(et1.getText().toString());
        ComplexNum c2 = new ComplexNum(et2.getText().toString());

        TextView tv = (TextView)findViewById(R.id.textView5);
        tv.setText(c1.divide(c2).toString());
    }

}

File: complexcalcKotlin/MainActivity.kt
package com.googleworkshop.jakub.complexcalc

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import ComplexNum

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun add(view: View) {
        val et1 = findViewById(R.id.editText) as EditText
        val et2 = findViewById(R.id.editText2) as EditText

        val c1 = ComplexNum(et1.text.toString())
        val c2 = ComplexNum(et2.text.toString())

        val tv = findViewById(R.id.textView5) as TextView
        tv.setText((c1 + c2).toString())
    }

    fun subtract(view: View) {
        val et1 = findViewById(R.id.editText) as EditText
        val et2 = findViewById(R.id.editText2) as EditText

        val c1 = ComplexNum(et1.text.toString())
        val c2 = ComplexNum(et2.text.toString())

        val tv = findViewById(R.id.textView5) as TextView
        tv.setText((c1 - c2).toString())
    }

    fun multiply(view: View) {
        val et1 = findViewById(R.id.editText) as EditText
        val et2 = findViewById(R.id.editText2) as EditText

        val c1 = ComplexNum(et1.text.toString())
        val c2 = ComplexNum(et2.text.toString())

        val tv = findViewById(R.id.textView5) as TextView
        tv.setText((c1 * c2).toString())
    }

    fun divide(view: View) {
        val et1 = findViewById(R.id.editText) as EditText
        val et2 = findViewById(R.id.editText2) as EditText

        val c1 = ComplexNum(et1.text.toString())
        val c2 = ComplexNum(et2.text.toString())

        val tv = findViewById(R.id.textView5) as TextView
        tv.setText((c1 / c2).toString())
    }

}
